server/modules/search/elasticsearch/engine.js

```diff
@@ -87,63 +84,156 @@ module.exports = {
-          suggest: {
-            suggestions: {
-              text: q,
-              completion: {
-                field: 'suggest',
-                size: 5,
-                skip_duplicates: true,
-                fuzzy: true
+          })
+          return {
+            results: _.get(results, 'body.hits.hits', []).map(r => ({
+              id: r._id,
+              locale: r._source.locale,
+              path: r._source.path,
+              title: r._source.title,
+              description: r._source.description,
+              createdAt: r._source.createdAt
+            })),
+            suggestions: _.reject(_.get(results, 'suggest.suggestions', []).map(s => _.get(s, 'options[0].text', false)), s => !s),
+            totalHits: _.get(results, 'body.hits.total.value', _.get(results, 'body.hits.total', 0))
+          }
+        } catch (err) {
+          WIKI.logger.warn('Search Engine Error: ', _.get(err, 'meta.body.error', err))
+        }
+      }else if (q.length ==15){
+        var exact_date = q.substr(5,10);
+        try {
+          const results = await this.client.search({
+            index: this.config.indexName,
+            body: {
+              query: {
+                range: {
+                  createdAt: {
+                    gte: exact_date,
+                    lte: exact_date
+                  }
+                }
+              },
+              suggest: {
+                suggestions: {
+                  text: q,
+                  completion: {
+                    field: 'suggest',
+                    size: 5,
+                    skip_duplicates: true,
+                    fuzzy: true
+                  }
+                }
               }
             }
+          })
+          return {
+            results: _.get(results, 'body.hits.hits', []).map(r => ({
+              id: r._id,
+              locale: r._source.locale,
+              path: r._source.path,
+              title: r._source.title,
+              description: r._source.description,
+              createdAt: r._source.createdAt
+            })),
+            suggestions: _.reject(_.get(results, 'suggest.suggestions', []).map(s => _.get(s, 'options[0].text', false)), s => !s),
+            totalHits: _.get(results, 'body.hits.total.value', _.get(results, 'body.hits.total', 0))
           }
+        } catch (err) {
+          WIKI.logger.warn('Search Engine Error: ', _.get(err, 'meta.body.error', err))
         }
-      })
-      return {
-        results: _.get(results, 'body.hits.hits', []).map(r => ({
-          id: r._id,
-          locale: r._source.locale,
-          path: r._source.path,
-          title: r._source.title,
-          description: r._source.description
-        })),
-        suggestions: _.reject(_.get(results, 'suggest.suggestions', []).map(s => _.get(s, 'options[0].text', false)), s => !s),
-        totalHits: _.get(results, 'body.hits.total.value', _.get(results, 'body.hits.total', 0))
       }
```

Comment:
You have 2 if with a condition but not else block. What happens if I type `date:abc`, it never returns.